File: temp/src/Game.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Game {
    private Board board;
    private Player player1;
    private Player player2;
    private Player currentPlayer;
    private Player winner = null;

    public Game(Player player1, Player player2) {
        this.board = new Board();
        this.player1 = player1;
        this.player2 = player2;
        this.currentPlayer = player1;
    }

    public Player getCurrentplayer() {
        return this.currentPlayer;
    }
    
    public Board getBoard() {
        return this.board;
    }

    public void switchCurrentPlayer() {
        if (this.currentPlayer.equals(this.player1)) {
            this.currentPlayer = this.player2;
        } else if (this.currentPlayer.equals(this.player2)) {
            this.currentPlayer = this.player1;
        }
    }

    public void display() {
        this.board.display();
    }

    public boolean isWinner() {
        return this.board.checkWinner();
    }

    public boolean isDraw() {
        return this.board.isFull();
    }

    public boolean isOver() {
        if (this.isDraw()) {
            this.winner = null;
            this.currentPlayer = null;
            return true;
        }

        if (this.isWinner()) {
            this.winner = this.currentPlayer;
            Utils.clearTerminal();
            this.board.setRemainingCells();
            this.display();
            return true;
        }

        return false;
    }

    public void announceDraw() {
        System.out.println("It is a draw!");
    }

    public void announceWinner() {
        System.out.printf("%s is the winner!\n", this.winner.getName());
    }

    public int[] getPlayerMove() throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));

        while (true) {
            System.out.printf("Player '%s', enter row and column (e.g. '1 2'): ", this.currentPlayer.getName());
            String input = reader.readLine();

            try {
                return Utils.parseInput(input, board);
            } catch (NumberFormatException nfe) {
                System.out.println(nfe.getMessage());
            }
        }
    }
}

File: tic_tac_py/src/game.py
from board import Board
from utils import clear_terminal, parse_input


class Game:
    def __init__(self, player1, player2):
        self.player1 = player1
        self.player2 = player2
        self.current_player = player1
        self.winner = None
        self.board = Board(3)

    def get_current_player(self):
        return self.current_player

    def get_board(self):
        return self.board

    def switch_current_player(self):
        self.current_player = self.player2 if self.current_player == self.player1 else self.player1

    def display(self):
        self.board.display()

    def is_winner(self):
        return self.board.check_winner()

    def is_draw(self):
        return self.board.is_full()

    def is_over(self):
        if self.is_draw():
            self.winner = None
            self.current_player = None
            return True

        if self.is_winner():
            self.winner = self.current_player
            clear_terminal()
            self.board.set_remaining_cells()
            return True

        return False

    def announce_draw(self):
        print("It is a draw!")

    def announce_winner(self):
        print(f"{self.winner.get_name()} is the winner!")

    def get_player_move(self):
        while True:
            input_str = input("Player '{}', enter row and column (e.g. '1 2'): ".format(
                self.current_player.get_name()))
            try:
                return parse_input(input_str, self.board)
            except ValueError as ve:
                print(ve)
